libres/lib/include/ert/enkf/row_scaling.hpp

```diff
@@ -1,67 +1,21 @@
-/*
-  Copyright (C) 2020  Equinor ASA, Norway.
-
-  The file 'row_scaling.hpp' is part of ERT - Ensemble based Reservoir Tool.
-
-  ERT is free software: you can redistribute it and/or modify
-  it under the terms of the GNU General Public License as published by
-  the Free Software Foundation, either version 3 of the License, or
-  (at your option) any later version.
-
-  ERT is distributed in the hope that it will be useful, but WITHOUT ANY
-  WARRANTY; without even the implied warranty of MERCHANTABILITY or
-  FITNESS FOR A PARTICULAR PURPOSE.
-
-  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
-  for more details.
-*/
+#pragma once
```

Comment:
Most of our code uses header guards.
Should we use pragma once instead?
My knowledge of the difference between the two is based entirely on the first paragraph of pragma onces' wiki page!

Reply:
Yeah I think it's the less confusing of the two. Consider if you're not careful and have two `.hpp` with the same name you might end up with the same header guard.

It'll be in writing at some point.

Reply:
From wiki:

"On the other hand, #pragma once is not necessarily available in all compilers and its implementation is tricky and might not always be reliable."

Is this something to worry about?

Reply:
It is available in all mainstream compilers and thus is both already implemented and reliable.

This is in reference to any of the hundreds of random half-baked C++ compilers that chip manufacturers make for their platforms.

Reply:
As in, the portability concerns that the C++ standard has, the compiler and we have are 3 entirely different beasts.

The C++ standard has to consider every single possible architecture, real or imagined (eg. there is hardware where allocating memory is not a thing, therefore large parts of the C++ standard library are designed to work without allocating memory. Another example is `NULL` not being the integer value `0` on some architectures, which is one of the reasons that C++11 introduced `nullptr`).

The compiler has to support a bajillion different operating systems and CPU architectures, including really outdated ones (eg. modern GCC can produce assembly compatible with the Intel 80368, first manufactured in 1985).

We only have the following: `(Linux + macOS) x (x86_64 + aarch64)`.

Reply:
Thanks for the nice write-up!